horovod/common/gradient_aggregation_eager.py

```diff
@@ -74,12 +74,17 @@ def compute_gradients(self, grads):
     def _allreduce_helper(self, grads):
         allreduced_grads = self._allreduce_grads(grads)
         if tf.math.greater(self.aggregation_frequency, 1):
-            gradient_divisor = (
-                self.aggregation_frequency if self.average_aggregated_gradients else 1
-            )
-            allreduced_grads = [
-                grad / gradient_divisor if grad is not None else None for grad in grads
-            ]
+            try:
+                gradient_divisor = (
+                    self.aggregation_frequency if self.average_aggregated_gradients else 1
+                )
+                allreduced_grads = [
+                    grad / gradient_divisor if grad is not None else None for grad in grads
+                ]
+            except Exception as e:
+                print(f"Warning: exception {e} generated during aggregation division. If you are using "
+                      "Indexed slices this is expected. If you are using aggregation of 1 this exception"
+                      "will have no effect.")
```

Comment:
>  Indexed slices this is expected. If you are using aggregation of 1 this exception will have no effect.

question: this seems a little odd? Can we instead gate this with some if statements and not catch exceptions?

Reply:
Took a new approach on this, after being able to reproduce the original error.